src/pages/unsubscribe.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import styled from 'styled-components'
+import { validate } from 'email-validator'
+
+import Section, { TopSection } from '../components/layout/Section'
+import { InputButton } from '../components/buttons/Button'
+import DefaultInput, { ErrorMessage } from '../components/form/Input'
+import Grid, { Col, Row } from '../components/layout/Grid'
+import { H2, H3, P } from '../components/text'
+import Header from '../components/layout/Header'
+import { WHITE, FONT_FAMILY } from '../config/styles'
+
+import { Label as DefaultLabel } from '../components/text'
+
+const Input = styled(DefaultInput)`
+  background-color: ${WHITE};
+`
+
+const Label = styled(DefaultLabel)``
+
+const ThanksTitle = styled(H3)`
+  margin: 1em 0;
+` //TODO: animate this later
+
+class Unsubscribe extends React.Component {
+  state = {
+    email: '',
+    formSubmited: false,
+    emailValid: false,
+  }
+
+  handleFormSubmit = e => {
+    e.preventDefault()
+    this.setState({ formSubmited: true })
+    this.triggerUnsubscribe()
+  }
+
+  triggerUnsubscribe = () => {
+    fetch(
```

Comment:
@ric9176 why don't you move the fetch into a different file, then add the fetch as a prop (https://github.com/reactjsacademy/fb-messenger/blob/testing-react-leanjs/src/components/Messenger/Conversation/Content/Messages.js#L125), and write a test for it? This looks like an important functionality

Reply:
Yes I agree, we should def do that :+1: was just being fast as this is the mvp implementation